linera-indexer/src/lib.rs

```diff
@@ -0,0 +1,146 @@
+// Copyright (c) Zefchain Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+//! This module defines the Linera indexer.
+
+pub mod application;
+pub mod applications;
+pub mod graphql;
+pub mod messages;
+pub mod operations;
+pub mod plugin;
+pub mod types;
+
+use crate::{
+    graphql::{block, Block},
+    types::IndexerError,
+};
+use async_recursion::async_recursion;
+use graphql_client::reqwest::post_graphql;
+use linera_base::{crypto::CryptoHash, data_types::BlockHeight, identifiers::ChainId};
+use linera_chain::data_types::HashedValue;
+use linera_views::{
+    common::{ContextFromDb, KeyValueStoreClient},
+    map_view::MapView,
+    value_splitting::DatabaseConsistencyError,
+    views::{RootView, View, ViewError},
+};
+use operations::OperationsPlugin;
+use std::{cmp::Ordering, sync::Arc};
+use tokio::sync::Mutex;
+use tracing::info;
+
+#[derive(RootView)]
+struct StateView<C> {
+    chains: MapView<C, ChainId, CryptoHash>,
+}
+
+pub struct Indexer<C> {
+    start: BlockHeight,
+    node: String,
+    state: Arc<Mutex<StateView<C>>>,
+    pub operations: Option<OperationsPlugin<C>>,
+}
+
+impl<DB> Indexer<ContextFromDb<(), DB>>
+where
+    DB: KeyValueStoreClient + Clone + Send + Sync + 'static,
+    DB::Error: From<bcs::Error>
+        + From<DatabaseConsistencyError>
+        + Send
+        + Sync
+        + std::error::Error
+        + 'static,
+    ViewError: From<DB::Error>,
+{
+    /// Loads all indexer plugins
+    pub async fn load(
+        plugins: &[&str],
```

Comment:
Do we print an error somewhere if an unknown plugin was specified?

Reply:
sofar no, but I haven't implemented the genericity of the plugins yet, the `operations` field in `Indexer` is not supposed to stay

Reply:
We could still make it fail for now if there's any plugin other than `operations`?